src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -183,11 +186,11 @@ protected internal bool AcceptIf(CSharpKeyword keyword)
             return false;
         }
 
-        protected static Func<CSharpToken, bool> IsSpacingToken(bool includeNewLines, bool includeComments)
+        protected static Func<SyntaxToken, bool> IsSpacingToken(bool includeNewLines, bool includeComments)
         {
-            return token => token.Type == CSharpTokenType.WhiteSpace ||
-                          (includeNewLines && token.Type == CSharpTokenType.NewLine) ||
-                          (includeComments && token.Type == CSharpTokenType.Comment);
+            return token => token.Kind == SyntaxKind.Whitespace ||
+                          (includeNewLines && token.Kind == SyntaxKind.NewLine) ||
+                          (includeComments && token.Kind == SyntaxKind.Comment);
```

Comment:
`Comment` might be a bit overloaded of a name given we have the same concept in Razor and HTML. It might be good to have a more specific name like `CSharpComment` or `SingleLineComment`. Also does this comment also represent multi-line C# comment?